### ISSUE TITLE
No ports with python-rtmidi v1.5.3 & python v3.11 on ARM Macs

### ISSUE THREAD

**Issue by Jasonhead44** [mido/mido#486]
Python-rtmidi v1.5.3 provides support for Python3.11 and arm based machines. After updating to this version mido functions no longer return the expected results.

get_input_names() returns a blank list when using python-rtmidi v1.5.3.

Reverting back to Python3.10 and python-rtmidi v1.4.9 fixes any issues and restores functionality.

I was unable to determine if this was an issue with the python-rtmidi dependency, so I thought it would be best to make an issue here as well in case there is an underlying bug in its implementation with mido.

I am relatively new to contributing on github and I apologize if this post is not detailed enough. I am happy to answer any questions that I can or provide any additional information.

**Comment by rdoursenaud** [comment]
Thanks for the report!

Sounds very similar to https://github.com/SpotlightKid/python-rtmidi/issues/150.

May I ask on which platform you are?

I haven't been able to reproduce this in Microsoft Windows 11 but I've encountered another while trying:
I don't get any MIDI messages from my ports neither in `polling` nor in `callback` modes even though they seem to be enumerated properly.

**Comment by rdoursenaud** [comment]
Tried all 1.5.x versions under Microsoft Windows 11:
- v1.5.0: exhibits not enumerating the ports properly
- v1.5.1 to 1.5.3: doesn't pass messages

I'll have to investigate more later.

**Comment by rdoursenaud** [comment]
This seem only to happen with Python 3.11.
Using v1.5.3 under Python 3.10 yields expected results.
Weird.

**Comment by rdoursenaud** [comment]
Nevermind.
I switched my app to its `DEBUG` mode and the MIDI messages shows in my log. I've hit an unrelated GUI bug.
I have no issues running `mido` under Python 3.11 with `python-rtmidi` version 1.5.3 under Microsoft Windows 11!

**Comment by rdoursenaud** [comment]
Thanks.

I unfortunately don't have an arm64 Mac in my lab yet to try and reproduce the issue.

This is probably an upstream bug.
Let's first make sure the problem is indeed in `python-rtmidi`.

Please run the following command from a shell (Running Python 3.11 and with python-rtmidi 1.5.3 installed from pip) and report its results:
```sh
python -c 'import rtmidi; print(rtmidi.get_compiled_api()); mi = rtmidi.MidiIn(); mo = rtmidi.MidiOut(); print(mo.get_ports()); print(mo.get_ports());'
```
If the ports lists are empty, the issue definitely is in `python-rtmidi`.
I'd appreciate that you report it [there](https://github.com/SpotlightKid/python-rtmidi/issues/) while linking to this issue to allow us to track resolution.

Else, we'll need to investigate more on our side.

**Comment by Jasonhead44** [comment]
I ran the command using Python 3.11.4, python-rtmidi 1.5.3 and mido 1.2.10:

Command:
``` bash
python3 -c 'import rtmidi; print(rtmidi.get_compiled_api()); mi = rtmidi.MidiIn(); mo = rtmidi.MidiOut(); print(mi.get_ports()); print(mo.get_ports());'
```
Results:
``` bash
[6]
[]
[]
```
I will report the issue under python-rtmidi and link back to this issue. Thank you for your help!

**Comment by rdoursenaud** [comment]
Thanks, this confirms that mido is out of the picture.
I'll keep the issue open until resolved upstream.

**Comment by rdoursenaud** [comment]
Seems to have been fixed in python-rtmidi `1.5.4`.
